07_errors/nicholascross/main.go

```
@@ -0,0 +1,169 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sync"
+)
+
+type Puppy struct {
+	ID     int64
+	Breed  string
+	Colour string
+	Value  float64
+}
+
+type Storer interface {
+	CreatePuppy(p Puppy) (bool, error)
+	RetrievePuppy(int64) (*Puppy, error)
+	UpdatePuppy(int64, Puppy) (bool, error)
+	DeletePuppy(int64) (bool, error)
```

Having both a `bool` and `error` return value seems redundant. Whenever the `bool` is `true`, `error` is `nil` and vice versa. You can just return an error, and all is good if it is `nil`.

This applies to `CreatePuppy`, `UpdatePuppy` and `DeletePuppy`